### PR TITLE
Cross-compilation fixes and improvements

### PR DESCRIPTION
**Pull request by Rbiessy**

### CODE REVIEW

File: third_party/sycl/crosstool/CROSSTOOL.tpl [hunk 1]
```diff
@@ -84,7 +84,6 @@ toolchain {
   unfiltered_cxx_flag: "-D__TIME__=\"redacted\""
   unfiltered_cxx_flag: "-no-canonical-prefixes"
 
-  compiler_flag: "-no-serial-memop"
```
Comment (DuncanMcBain) [inline third_party/sycl/crosstool/CROSSTOOL.tpl:85]: Are we definitely removing this again? Feels like we've just added it, and it's (fairly) portable.

File: third_party/sycl/crosstool/CROSSTOOL.tpl [hunk 2]
```diff
@@ -192,9 +191,9 @@ toolchain {
   unfiltered_cxx_flag: "-D__TIME__=\"redacted\""
   unfiltered_cxx_flag: "-no-canonical-prefixes"
 
-  compiler_flag: "-U_FORTIFY_SOURCE"
-  compiler_flag: "-D_FORTIFY_SOURCE=1"
-  compiler_flag: "-fstack-protector"
+  compiler_flag: "-DDISABLE_SKINNY=1"
```
Comment (DuncanMcBain) [inline third_party/sycl/crosstool/CROSSTOOL.tpl]: We should add this the the bzl_configure, as we don't always want to disable this. Ideally the crosstool shouldn't change between branches.

File: third_party/sycl/crosstool/CROSSTOOL.tpl (hunk 1 above)
Comment (Rbiessy) [reply third_party/sycl/crosstool/CROSSTOOL.tpl:85]: I should have removed it only for cross-compilation. Now it actually makes more sense to have this option in all of our configs except sycl_arm

Comment (alistair-low) [reply third_party/sycl/crosstool/CROSSTOOL.tpl:85]: My understanding of -no-serial-memop is that it removes a workaround that we need for some buggy OpenCL drivers (on both desktop and embedded devices). If we have it set by default then we will cause problems for users that follow our guides to build and run TensorFlow on such an OpenCL device, and the errors that they see will not be obviously related to this flag (as far as I know). So I believe that it should be removed here and also from the bazel.rc. Users can specify it in their build if they already verified that their build works without it

Comment (Rbiessy) [reply third_party/sycl/crosstool/CROSSTOOL.tpl:85]: Ok thanks for the info. We removed it from our scripts and will add the option only on the command line when needed.

**Pull request merged**